velox/exec/PrefixSortAlgorithm.cpp

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/exec/PrefixSortAlgorithm.h"
+
+namespace facebook::velox::exec {
+
+PrefixSortContext::PrefixSortContext(
+    memory::MemoryPool* pool,
+    uint64_t entrySize,
+    char* end)
+    : entrySize_(entrySize),
+      swapBuffer_(AlignedBuffer::allocate<char>(entrySize, pool)),
```

Comment:
If memory pool is used only to allocate the buffer, perhaps, we can change the API to require the caller to provide the buffer to use. This would remove dependency on memory pool.

```
PrefixSortContext(
    uint64_t entrySize,
    char* end,
    BufferPtr& swapBuffer)
```

Reply:
The purpose to keep pool is that : maybe we need to allocate multi buffer to avoid data dependency, and alos need more buffer for simd case. So , could we keep the pool ?

Reply:
>I wonder if it makes sense to move it into a separate directory though. Maybe exec/prefixsort?

Yes , I aggree with this~